FirestoreDocument.js

```diff
@@ -66,40 +66,79 @@ function wrapValue_ (value) {
 
 function unwrapValue_ (value) {
   var type = Object.keys(value)[0]
+  value = value[type]
   switch (type) {
+    case 'referenceValue':
+      // TODO: Allow possibility to retrieve document from this reference
+      // return value.match(/\/documents\/(.*)/)[1];
+      // fall through
+    case 'bytesValue':
     case 'stringValue':
     case 'booleanValue':
     case 'integerValue':
```

Comment:
This isn't an issue with your changes, but here's a bug I just discovered while testing your stuff out: when an integer gets parsed here, it comes out as a string. Here's a test function for observing this:

```javascript
function test() {
  const collectionName = "numbers_to_strings";
  
  const numberDoc = {
    "integer": 120,
    "double": 1.20
  }
  
  firestore.createDocumentWithId("first", collectionName, numberDoc);

  var storedDoc = firestore.getDocument(collectionName + "/first").fields
  
  // The locally created object has numbers for fields:
  Logger.log(numberDoc);
  
  // The one we read from Firestore has its integer converted to a string:
  Logger.log(storedDoc);
  
  // If we send these fields to Firestore, the integer field is now a string:
  firestore.createDocumentWithId("second", collectionName, storedDoc)
  
  var doubleStored = firestore.getDocument(collectionName + "/second");
  Logger.log(doubleStored);
}
```

It's an easy fix — just use `parseInt(...)` when unwrapping an integer instead of directly returning its value. That is, in `unwrapValue`'s integer case, do this:

```javascript
case 'integerValue':
      return parseInt(value)
```

This is a separate issue from what you're addressing, but since you're editing this part of the code, would you mind inserting this into your PR so we don't have to deal with any merge conflicts?